### PR TITLE
[SDK/CLI] Add private function to allow user add run properties for local to cloud

### PR DESCRIPTION
**Pull request by 0mza987**
# Description

- Add private function to allow user add run properties for local to cloud.
- Add `total_tokens` to run properties when uploading to cloud.

Currently can only set `_azureml.evaluation_run` and `_azureml.evaluate_artifacts`.

```python
  # submit a local batch run
  run = pf._run(
      flow=f"{FLOWS_DIR}/simple_hello_world",
      data=f"{DATAS_DIR}/webClassification3.jsonl",
      name=name,
      column_mapping={"name": "${data.url}"},
      display_name="sdk-cli-test-run-local-to-cloud-with-properties",
      tags={"sdk-cli-test": "true"},
      description="test sdk local to cloud",
      properties={
          "_azureml.evaluation_run": eval_run,
          "_azureml.evaluate_artifacts": eval_artifacts,
      },
  )
```

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by Adrian999999999 — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_sdk/entities/_run.py [hunk 1]
```diff
@@ -661,6 +662,15 @@ def _to_rest_object(self):
             start_time = self._created_on.isoformat() + "Z" if self._created_on else None
             end_time = self._end_time.isoformat() + "Z" if self._end_time else None
 
+            # extract properties that needs to be passed to the request
+            total_tokens = self.properties[FlowRunProperties.SYSTEM_METRICS].get("total_tokens", 0)
+            properties = {
+                Local2CloudProperties.TOTAL_TOKENS: total_tokens,
+                Local2CloudProperties.EVAL_RUN: self.properties.get(Local2CloudProperties.EVAL_RUN, None),
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/entities/_run.py:669]: maybe we can have a constants of all the accepted properties list, and handle it more generally?

File: src/promptflow-devkit/promptflow/_sdk/_pf_client.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ def run(
         variant: str = None,
         connections: dict = None,
         environment_variables: dict = None,
+        properties: dict = None,
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_pf_client.py:75]: @singankit Please use this _run() function which exposes the properties parameter. We hide it from the public API.

File: src/promptflow-devkit/promptflow/_sdk/entities/_run.py (hunk 1 above)
Comment (0mza987) [reply src/promptflow-devkit/promptflow/_sdk/entities/_run.py]: Updated.

File: src/promptflow-devkit/promptflow/_sdk/_pf_client.py (hunk 2 above)
Comment (0mza987) [reply src/promptflow-devkit/promptflow/_sdk/_pf_client.py:75]: Updated the usage in pr description

**Review by wangchao1230 — APPROVED** [review]

**Review by YingChen1996 — APPROVED** [review]

**Pull request merged**